src/Hosting/Hosting/src/Internal/HostingApplication.cs

```diff
@@ -78,6 +78,7 @@ public Context CreateContext(IFeatureCollection contextFeatures)
             }
 
             _diagnostics.BeginRequest(httpContext, hostContext);
+            httpContext.Features.Set<IActivityFeature>(new ActivityFeature());
```

Comment:
Should the feature be "known" for perf? Avoid allocation of creating feature and dictionary overhead of an unknown feature.

Reply:
Yes